### ISSUE TITLE
UTF-8 conversion bug

### ISSUE THREAD

**Issue by kwerle** [arton/rjb#97]
There seems to be a UTF-8 conversion bug introduced 4 years ago in 1.6.0.  Our Korean test string is:

```
        -"국군은 국가의 안전보장과 국\u{78029}위의 신성\u{6731D}�무를 수\u{72568}을 사명으로 \u{66270}."
        +"국군은 국가의 안전보장과 국토방위의 신성한 의무를 수행함을 사명으로 하며."
```
When migrating the + value it becomes the - value and should not.  I will attempt to write a trivial test.

**Comment by kwerle** [comment]
Here is the outline of a trivial test:
```
test_string = "국군은 국가의 안전보장과 국토방위의 신성한 의무를 수행함을 사명으로 하며."    
JavaStringClass = Rjb::import('java.lang.String')
string_from_java = JavaStringClass.new_with_sig('Ljava.lang.String;', test_string).toString
test_string == string_from_java     # true in 1.5.9, false in 1.6
pp [test_string, string_from_java]
```

**Comment by bjonnh-work** [comment]
The problem seem to be related the changes here:
https://github.com/arton/rjb/blame/master/ext/riconv.c#L170

Looks like it has trouble with characters like: 
`53664 | 0xD1A0 | 토 | HANGUL SYLLABLE TO`

**Comment by arton** [comment]
@kwerle @bjonnh-work Thanks for your bug report and a simple reproduce string.
I'd fixed the issue and push rjb-1.6.9.gem to RubyGems.
cf) test/test.rb#test_java_hangul_syllable.

**Comment by kwerle** [comment]
@arton Thank you for all your wonderful work!
